examples/org.eclipse.cdt.lsp.examples.preferences/OSGI-INF/org.eclipse.cdt.lsp.examples.preferences.MyClangdOptionsDefaults.xml

```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.3.0" name="org.eclipse.cdt.lsp.examples.preferences.MyClangdOptionsDefaults">
+   <service>
+      <provide interface="org.eclipse.cdt.lsp.clangd.ClangdOptions"/>
+   </service>
+   <property name="service.ranking" type="Integer" value="100"/>
+   <property name="isDefaultOptions" type="Boolean" value="true"/>
```

Comment:
`org.eclipse.cdt.lsp.internal.clangd.ClangdMetadataDefaults.xml` does. This was my attempt to do something similar to what I usually do with named bindings, as I know them from google guice. I did this to avoid having to create another interface for the ClangdOptions defaults.

Reply:
Ah I missed this `target="(isDefaultOptions=true)"`
Usually I prefer to specify everything using `@Component` annotation attributes and then PDE DS builder supports xml manifest for me.

Reply:
@ruspl-afed shouldn't we also specify the service in the `@Component` annotation then, e.g. change

```Java
@Component
public final class ClangdFallbackManager implements ClangdFallbackFlags {
```

to
```Java
@Component(service = ClangdFallbackFlags.class)
public final class ClangdFallbackManager implements ClangdFallbackFlags {
```

Reply:
> @ruspl-afed shouldn't we also specify the service in the `@Component` annotation then, e.g. change
> 
> ```java
> @Component
> public final class ClangdFallbackManager implements ClangdFallbackFlags {
> ```
> 
> to
> 
> ```java
> @Component(service = ClangdFallbackFlags.class)
> public final class ClangdFallbackManager implements ClangdFallbackFlags {
> ```

Okay, seems like the Tooling does this automatically, never mind.

Reply:
DS Builder does it for interfaces mentioned in signature, usually this is what was desired, but not always.